engine/src/flutter/lib/web_ui/lib/src/engine/semantics/dialogs.dart

```diff
@@ -0,0 +1,53 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'label_and_value.dart';
+import 'semantics.dart';
+import 'route.dart';
+
+/// Indicates the container as a pop dialog
+///
+/// Uses aria dialog role to convey this semantic information to the element.
+///
+/// Setting this role will also set aria-modal to true, which helps screen
+/// reader better understand this section of screen.
+///
+/// Screen-readers takes advantage of "aria-label" to describe the visual.
+class SemanticDialog extends SemanticRole with RouteLike {
```

Comment:
Might be simpler to maintain all route semantics in the same file, `route.dart`. WDYT?